### PR TITLE
Add initial implementation of OPDRs

### PR DESCRIPTION
**Pull request by maxwofford**

### CODE REVIEW

File: app/views/organizer_position_deletion_requests/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        <td><%= opdr.submitted_by.name %></td>
+        <td><%= opdr.organizer_position.user.name %></td>
+        <td><%= opdr.status.to_s.humanize %></td>
+        <td><%= link_to (opdr.under_review? ? 'Process' : 'View'), opdr %></td>
+      </tr>
+    <% end %>
+  </tbody>
+</table>
+<% else %>
+<h3>Everything is well in the land of bank</h3>
```
Comment (tmb) [inline app/views/organizer_position_deletion_requests/index.html.erb]: capital b

File: app/views/organizer_position_deletion_requests/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+<% if @opdr.organizer_position.deleted_at.nil? %>
+<%= link_to 'Delete organizer ⚠️☣☠', organizer_path(@opdr.organizer_position.id), method: :delete, data: {confirm: "This will remove this user from #{@opdr.organizer_position.event.name} from the database. Are you sure you want to do this?"}, class: 'btn bg-error ml1' %>
+<% else %>
+<p><strong>Organizer is marked as deleted</strong></p>
+<% end %>
+
+<% if @opdr.under_review? %>
+<%= link_to 'Close Request', organizer_position_deletion_request_close_path(@opdr.id), method: :post, data: {confirm: "This will mark the request as processed. Only continue if no follow-up is needed for this request."}, class: 'btn bg-accent ml1' %>
+<% else %>
+<%= link_to 'Open Request', organizer_position_deletion_request_open_path(@opdr.id), method: :post, data: {confirm: "This will mark the request as processed. Only continue if no follow-up is needed for this request."}, class: 'btn bg-info ml1' %>
```
Comment (tmb) [inline app/views/organizer_position_deletion_requests/show.html.erb]: doesn't open mean that it will be unprocessed again?

**Review by tmb — DISMISSED** [review]
see comments

File: app/controllers/organizer_position_deletion_requests_controller.rb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    redirect_to @opdr, notice: 'Removal request closed.'
+  end
+
+  def open
+    authorize OrganizerPositionDeletionRequest
+    @opdr.closed_at = nil
+    @opdr.closed_by = nil
+    @opdr.save
+
+    redirect_to @opdr, notice: 'Removal request openned.'
```
Comment (tmb) [inline app/controllers/organizer_position_deletion_requests_controller.rb]: typo

File: app/models/organizer_position_deletion_request.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  belongs_to :organizer_position, -> { with_deleted }
+
+  has_many :comments, as: :commentable
+
+  scope :under_review, -> { where(closed_at: nil) }
+
+  validates_presence_of :reason
+
+  def under_review?
+    closed_at == nil
```
Comment (lachlanjc) [inline app/models/organizer_position_deletion_request.rb]: .nil?

File: app/models/organizer_position_deletion_request.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def status
+    if closed_at == nil
```
Comment (lachlanjc) [inline app/models/organizer_position_deletion_request.rb]: Can use a ternary for brevity

File: app/views/organizer_position_deletion_requests/index.html.erb [hunk 6]
```diff
@@ -0,0 +1,32 @@
+<p id="notice"><%= notice %></p>
```
Comment (lachlanjc) [inline app/views/organizer_position_deletion_requests/index.html.erb]: Eek, pop this out :)

File: app/views/organizer_position_deletion_requests/show.html.erb [hunk 7]
```diff
@@ -0,0 +1,50 @@
+<p id="notice"><%= notice %></p>
+
+<p>
```
Comment (lachlanjc) [inline app/views/organizer_position_deletion_requests/show.html.erb]: Wrap this in div.details for nice styling (& axe the notice)

File: app/controllers/organizer_position_deletion_requests_controller.rb [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+    attributes[:submitted_by] = current_user
+    @opdr = OrganizerPositionDeletionRequest.new(attributes)
+    @op = @opdr.organizer_position
+    @target = @op.user
+    @event = @op.event
+
+    authorize @opdr
+
+    if @opdr.save
+      redirect_to @event, notice: 'Removal request accepted. We’ll be in touch shortly.'
```
Comment (lachlanjc) [inline app/controllers/organizer_position_deletion_requests_controller.rb]: Use flash[:success]

File: app/controllers/organizer_position_deletion_requests_controller.rb [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    else
+      render :new
+    end
+  end
+
+  def close
+    authorize OrganizerPositionDeletionRequest
+    @opdr.closed_at = Time.now
```
Comment (lachlanjc) [inline app/controllers/organizer_position_deletion_requests_controller.rb]: Should this be moved to the model?

File: app/views/organizer_position_deletion_requests/show.html.erb (hunk 2 above)
Comment (maxwofford) [reply app/views/organizer_position_deletion_requests/show.html.erb]: Yep. This is just so humans have control over the process (ie. I'm an admin & I accidentally close the request before I should have)

Comment (tmb) [reply app/views/organizer_position_deletion_requests/show.html.erb]: yes, but look at the confirm ;)

(this was my bad, I should've worded it explicitly about the confirm)

Comment (maxwofford) [reply app/views/organizer_position_deletion_requests/show.html.erb]: Ah, gotcha. Just fixed!

**Review by lachlanjc — APPROVED** [review]
Haven't tried any of this locally but looks great!

**Comment by maxwofford** [comment]
I covered this with @MDestefanis11 & got his 👌. @lachlanjc: would really appreciate it if you can work your magic on the generated pages

**Pull request merged**